### ISSUE TITLE
ghgvc-dl deployment not working

### ISSUE THREAD

**Issue by dlebauer** [ebimodeling/ghgvc#63]
This may have been messed up by changing owner of the directory from 'nobody' to 'hmb'

**Comment by dlebauer** [comment]
from @gsrohde (with revisions for clarity):


Here are the details about what I have set up for @hmb1 :

1. The files are at /usr/local/hmb-ghgvc/ghgvc and /usr/local/hmb-ghgvc/ghgvcR.
2. important: I didn’t manage to get things working with the usual sub-URI configuration in Apache/Passenger.  So, for now at least (and let me know if you want me to try working on this further), I have deployed to virtual host www.ecosystemservicescalc.hmb.  Since this isn’t a URL recognized by any DNS server, Henry will have to add an entry in his hosts file to be able to pull this up in his browser.  The entry should be a line with just "128.174.127.81 www.ecosystemservicescalc.hmb” (without quotes).  On a Mac, this file is at /private/etc/hosts or just /etc/hosts.
3. important: I had to back up to a previous git version (closer to—in fact, almost the same as--the one running at /usr/local/ghgvc) in order to get things to work.  I made a git branch at this version and called it ‘last-working-version’.  It is actually on that branch now, so if Henry does any work and checks it in, it will be on that branch unless he changes it.  If you want me too, I can try to debug the problem(s) with the more recent versions.
4. important: I made one change to one file under /usr/local/hmb-ghgvc/ghgvc: Namely, I changed the value for ghgvcR_instantiation_path in the workflows controller from “/usr/local/ghgvcR” to “#{Rails.root}/../ghgvcR”.  The former works, but it uses the wrong copy of ghgvcR.  I’m assuming that you don’t want to share this one instance of ghgvcR among various deployments, though I suppose you could, especially if Henry is only going to be working on the Rails code and not the R code.  We may want to check this change in to make ghgvc always use the copy of ghgvcR in the sibling directory.  (Your copy at /usr/local/dl-ghgvc/ghgvc, for instance, uses the R code at /usr/local/ghgvcR.)
5. I had to use curl and yum to upgrade our copy of nodejs to get the assets to compile without error.  I thought I had precomplied them for the copy at /usr/local/ghgvc when we first deployed there, but it’s possible I just copied them from the Ubuntu version.  More likely, I think, is that the older nodejs wasn’t compatible with newer versions of some of the Ruby gems were are using, or with the newer version of Passenger we are using.
6. I had to configure Bundler to ignore the development environment to avoid the error about ruby-netcdf not being installed.  The other deployments were similarly configured so I figured that Gem wasn’t needed.

Was your copy (the one at /usr/local/dl-ghgvc/) ever working fully?  If so, it means the sub-URL deployment should work.  Did you set this up or did I?

@hmb1 - please create a new issue if into further deployment-related problems.  (At a minimum, it would be useful to know that PhusionPassenger by default runs as a different user that Apache does—generally it runs as the user who owns the file config.ru in the Rails root directory.)  Also, I could write up the steps I used to make this new deployment if that would be useful.

if you are going to be working on JavaScript code, there are certain changes that could be made to the deployment that would make debugging easier.  Please create an issue and assign it to @gsrohde. Otherwise, I’d point him to the Rails Guide on the Asset Pipeline if you want to delve into this yourself.

**Comment by dlebauer** [comment]
@hmb1 can you please check that the server is running as expected? As mentioned above, if you have further issues, please create a new issue and assign to either me or @gsrohde

**Comment by gsrohde** [comment]
UPDATE: I changed the hmb deployment to run in the development environment.  This will allow @hmb1 to see changes simply by refreshing the browser (including Javascript changes) instead of having to restart passenger or apache and/or clear the cache or recompile assets.  In fact, I removed the compiled assets (public/assets) also, since they aren't needed in development mode.

I also got the dl-ghgvc copy working at virtual host www.ecosystemservicescalc.dl (again, you'll have to add it to your hosts file to see it in your browser).  I had to back up the git version to fd89af (same as the hmb copy), and again, I put it on a branch called 'last-working-version'.  In both cases, these are just local branches.  I also marked the version the dl copy _was_ at (74ada8) as branch 'davids-former-version' (also just local) just in case you ever need to know where it was before I reverted changes.  And I switched this copy to run in development mode as well.

I don't know why I had trouble with subURI deployment (e.g., making the copy available at pecandev.igb.illinois.edu/dl-ghgvc) if it really was working before.  I did, however, find some ways to get this working.  But I will not pursue this further right now unless someone has a really problem with using the "hosts file"/"virtual hosting" method.
